frontend/src/components/manageUsers/RoleChangeModal.tsx

```diff
@@ -0,0 +1,70 @@
+"use client";
+
+import { useTranslations } from "next-intl";
+import React, { RefObject } from "react";
+import {
+  Button,
+  ButtonGroup,
+  ModalFooter,
+  ModalRef,
+  ModalToggleButton,
+} from "@trussworks/react-uswds";
+
+import { LoadingButton } from "src/components/LoadingButton";
+import { SimplerModal } from "src/components/SimplerModal";
+
+export interface RoleChangeModalProps {
+  isSubmitting: boolean;
+  modalRef: RefObject<ModalRef | null>;
+  nextRoleName: string;
+  onConfirm: () => void;
+  onCancel: () => void;
+}
+
+export function RoleChangeModal({
+  isSubmitting,
+  modalRef,
+  nextRoleName,
+  onConfirm,
+  onCancel,
+}: RoleChangeModalProps) {
+  const t = useTranslations("ManageUsers.confirmationModal");
+  const modalTitle = t("header");
+  const modalDescription = `${t("description")} ${nextRoleName}?`;
+
+  return (
+    <SimplerModal
+      modalId="confirmation-role-change-modal"
+      modalRef={modalRef}
+      titleText={modalTitle}
+      className="text-wrap"
```

Comment:
I think you need to pass `onCancel` to `onClose` here to fix the bug I mentioned last time
```suggestion
      className="text-wrap"
      onClose={onCancel}
```